tests/python/pants_test/cache/test_s3_artifact_cache.py

```diff
@@ -0,0 +1,198 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+from contextlib import contextmanager
+
+import boto3
+import pytest
+from moto import mock_s3
+
+from pants.cache.artifact_cache import UnreadableArtifact
+from pants.cache.local_artifact_cache import LocalArtifactCache, TempLocalArtifactCache
+from pants.cache.s3_artifact_cache import S3ArtifactCache
+from pants.invalidation.build_invalidator import CacheKey
+from pants.util.contextutil import temporary_dir, temporary_file
+from pants.util.dirutil import safe_mkdir
+
+
+_TEST_CONTENT1 = b'fraggle'
+_TEST_CONTENT2 = b'gobo'
+
+_TEST_BUCKET = 'verst-test-bucket'
+
+
+@pytest.yield_fixture(scope="function")
```

Comment:
Nit: single quotes, here and all other fixtures below. Although I guess it doesn't matter since you're going to rewrite this anyway.